libxrdp/libxrdp.c

```diff
@@ -373,43 +375,59 @@ libxrdp_send_bell(struct xrdp_session *session)
     return 0;
 }
 
-
 /*****************************************************************************/
 int EXPORT_CC
 libxrdp_send_bitmap(struct xrdp_session *session, int width, int height,
                     int bpp, char *data, int x, int y, int cx, int cy)
 {
-    int line_size = 0;
+    int line_bytes = 0;
     int i = 0;
     int j = 0;
+    int k;
     int total_lines = 0;
     int lines_sending = 0;
     int Bpp = 0;
     int e = 0;
     int bufsize = 0;
     int total_bufsize = 0;
     int num_updates = 0;
+    int line_pad_bytes;
+    int server_line_bytes;
     char *p_num_updates = (char *)NULL;
     char *p = (char *)NULL;
     char *q = (char *)NULL;
     struct stream *s = (struct stream *)NULL;
     struct stream *temp_s = (struct stream *)NULL;
+    tui32 pixel;
 
-    DEBUG(("libxrdp_send_bitmap sending bitmap"));
+    LLOGLN(10, ("libxrdp_send_bitmap: sending bitmap"));
     Bpp = (bpp + 7) / 8;
-    e = width % 4;
-
-    if (e != 0)
+    e = (4 - width) & 3;
+    switch (bpp)
     {
-        e = 4 - e;
+        case 15:
+        case 16:
+            server_line_bytes = width * 2;
+            break;
+        case 24:
+        case 32:
+            server_line_bytes = width * 4;
+            break;
+        default: /* 8 bpp */
+            server_line_bytes = width;
+            break;
```

Comment:
Please define `server_Bpp` (any reasonable name) and use `server_line_bytes = width * server_Bpp`